### PR TITLE
CCIP-4329 modify setChainRateLimiterConfig file for setting multiple rate limits

### PR DESCRIPTION
**Pull request by jhweintraub**
Update the `setChainRateLimitConfig()` function to now accept an array of configs to make front-end management of the token pool easier.

### CODE REVIEW

File: contracts/src/v0.8/ccip/pools/TokenPool.sol [hunk 1]
```diff
@@ -506,18 +506,20 @@ abstract contract TokenPool is IPoolV1, Ownable2StepMsgSender {
...
+  /// @param inboundConfigs The new inbound rate limiter config, meaning the offRamp rate limits for the given chain.
+  function setChainRateLimiterConfigs(
+    uint64[] calldata remoteChainSelectors,
+    RateLimiter.Config[] calldata outboundConfigs,
+    RateLimiter.Config[] calldata inboundConfigs
   ) external {
     if (msg.sender != s_rateLimitAdmin && msg.sender != owner()) revert Unauthorized(msg.sender);
 
-    _setRateLimitConfig(remoteChainSelector, outboundConfig, inboundConfig);
+    for (uint256 x = 0; x < remoteChainSelectors.length; x++) {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/pools/TokenPool.sol]: Any reason we use x? We always use i

Comment (jhweintraub) [reply contracts/src/v0.8/ccip/pools/TokenPool.sol]: not really. Oversight error.

File: contracts/src/v0.8/ccip/test/pools/TokenPool/TokenPool.setChainRateLimiterConfigs.t.sol [hunk 2]
```diff
@@ -51,20 +80,34 @@ contract TokenPool_setChainRateLimiterConfig is TokenPoolSetup {
   // Reverts
 
   function test_OnlyOwnerOrRateLimitAdmin_Revert() public {
+    uint64[] memory chainSelectors = new uint64[](1);
+    chainSelectors[0] = DEST_CHAIN_SELECTOR;
+
+    RateLimiter.Config[] memory newOutboundConfigs = new RateLimiter.Config[](1);
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/pools/TokenPool/TokenPool.setChainRateLimiterConfigs.t.sol:86]: I prefer the absolute minimal setup for reverts, meaning we can pass in empty lists here and remove most of this setup

File: contracts/src/v0.8/ccip/pools/TokenPool.sol [hunk 3]
```diff
@@ -506,18 +506,20 @@ abstract contract TokenPool is IPoolV1, Ownable2StepMsgSender {
...
+  /// @param inboundConfigs The new inbound rate limiter config, meaning the offRamp rate limits for the given chain.
+  function setChainRateLimiterConfigs(
+    uint64[] calldata remoteChainSelectors,
+    RateLimiter.Config[] calldata outboundConfigs,
+    RateLimiter.Config[] calldata inboundConfigs
   ) external {
     if (msg.sender != s_rateLimitAdmin && msg.sender != owner()) revert Unauthorized(msg.sender);
 
-    _setRateLimitConfig(remoteChainSelector, outboundConfig, inboundConfig);
+    for (uint256 i = 0; i < remoteChainSelectors.length; ++i) {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/pools/TokenPool.sol:554]: Lets add a length check to assert all 3 are the same length

File: contracts/src/v0.8/ccip/test/pools/TokenPool/TokenPool.setChainRateLimiterConfigs.t.sol [hunk 4]
```diff
@@ -51,20 +80,34 @@ contract TokenPool_setChainRateLimiterConfig is TokenPoolSetup {
...
+    s_tokenPool.setChainRateLimiterConfigs(chainSelectors, newOutboundConfigs, newInboundConfigs);
   }
 
   function test_NonExistentChain_Revert() public {
     uint64 wrongChainSelector = 9084102894;
 
+    uint64[] memory chainSelectors = new uint64[](1);
+    chainSelectors[0] = wrongChainSelector;
+
+    RateLimiter.Config[] memory newOutboundConfigs = new RateLimiter.Config[](1);
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/pools/TokenPool/TokenPool.setChainRateLimiterConfigs.t.sol:104]: Same for simplistic test setup here as well

**Review by 0xsuryansh — APPROVED** [review]

**Review by RensR — APPROVED** [review]

**Pull request merged**